swarmlib/util/functions.py

```diff
@@ -28,7 +31,19 @@ def ackley(x, a=20, b=0.2, c=2*np.pi):
     return -a*np.exp(-b*np.sqrt(s1 / n)) - np.exp(s2 / n) + a + np.exp(1)
 
 
+# Wrapper for landscapes.single_objective functions for inputs > 1d
+def wrap_landscapes_func(landscapes_func):
+    def wrapper(x):
+        return np.apply_along_axis(func1d=landscapes_func, axis=0, arr=x)
+    return wrapper
+
+
+# Add all functions from landscapes.single_objective
 FUNCTIONS = {
-    'michalewicz': michalewicz,
-    'ackley': ackley
+    name: wrap_landscapes_func(func)
+    for (name, func) in inspect.getmembers(
+        landscapes.single_objective, inspect.isfunction
+    )
 }
+# Replace / add functions that are defined in this file
+FUNCTIONS.update({'michalewicz': michalewicz, 'ackley': ackley})
```

Comment:
we are good to remove swarmlib's `michalewicz` and `ackley` functions since they are implemented in `landscapes` as well

Reply:
Initially that was my idea too. Playing around a little I got a bit confused about a few outputs of `michalewicz` (see below). I'd probably require your input on this one. If you want to use the `landscapes` implementation that's fine by me. I'd love to hear your take on this.

On a side note: Keeping `FUNCTIONS.update()` would allow users to define their own functions for CLI use quite easily.

```
>>> import landscapes.single_objective
>>> from functools import reduce
>>> import numpy as np
>>> 
>>> def michalewicz(x):
...     m = 10
...     def func(x):
...         return np.sin(x) * np.power(np.sin((0 + 1) * np.power(x, 2) / np.pi), 2 * m)
...     result = reduce(lambda acc, x: acc + func(x), x, 0.)
...     return -result
... 
>>> landscapes.single_objective.michalewicz((1,1))
-2.5573872831813936e-05
>>> michalewicz((1,1))
-1.3674002502623617e-10
>>> landscapes.single_objective.michalewicz((2.5,1.4))
-0.43900334157529114
>>> michalewicz((2.5,1.4))
-0.09832473095647655
>>> landscapes.single_objective.michalewicz((1.4,2.5))
-0.001587081586700395
>>> michalewicz((1.4,2.5))
-0.09832473095647655
```

Reply:
Hmmm maybe the implementation i submitted nathanrooy/landscapes#3 is faulty. 
This could also be because the function signatures are different, the implementation of landscapes takes the m parameter as argument. Let us check which one is the correct implementation 😅

Reply:
Hi @alxfmpl I just checked on the functions and found out that swarmlib's implementation of the michalewicz function is false. The term `(0 + 1)` is _not_ x_i like it should be according to the formula. Further the global minimum should be at `-1.8013` for `x=[2.20319,1.57049]`

```python
>>> from landscapes.single_objective import michalewicz
>>> from swarmlib.util.functions import FUNCTIONS
>>> FUNCTIONS['michalewicz']([2.20319,1.57049])
-0.8022726911732864
>>> michalewicz([2.20319,1.57049])
-1.8012982949924439
```

Therefore, you can remove swarmlib's functions and use landscape's implementations instead.

Reply:
Yep I reached the same conclusion. I'll remove swarmlib's functions.